internal/server/authn/oidc.go

```diff
@@ -192,7 +192,7 @@ func (o *oidcImplementation) GetUserInfo(providerUser *models.ProviderUser) (*Us
 		return nil, fmt.Errorf("decode user info response: %w", err)
 	}
 
-	if info.Groups == nil {
+	if info.Groups == nil || len(*info.Groups) == 0 {
```

Comment:
`*[]Groups` is an interesting type for this field. Usually pointer to slice isn't necessary because a slice itself can be nil. Do you know why we use this type of field?

If we used `[]Groups`, I think we'd only need to check `len(info.Groups) == 0`, since a nil slice also has zero length.

Reply:
Good call out, originally I was using a pointer because I only wanted to log a warning if there was no groups field on a response at all (rather than an empty groups field, which means the application was configured correctly, but the user wasn't in any groups). This update means we will log both of those cases so I changed it to a slice rather than a pointer to a slice.